.travis/install.sh

```diff
@@ -61,6 +61,28 @@ else
         pyenv install pypy-4.0.1
         pyenv global pypy-4.0.1
     fi
+    if [[ "${OPENSSL}" == "0.9.8" ]]; then
+      # download, compile, and install if it's not already present via travis cache
+      # This is 0.9.8l rather than zh because we have some branches for handling
+      # < 0.9.8m that won't be exercised with a newer OpenSSL. (RHEL5 is 0.9.8e with
+      # patches, but while that's in jenkins we don't get coverage data from it)
+      if [[ ! -f "$HOME/ossl-098l/bin/openssl" ]]; then
+        curl -O https://www.openssl.org/source/openssl-0.9.8l.tar.gz
+        tar zxvf openssl-0.9.8l.tar.gz
+        cd openssl-0.9.8l
+        echo "OPENSSL_0.9.8L_CUSTOM {
+            global:
+              *;
+        };" > openssl.ld
+        ./config no-asm no-ssl2 -Wl,--version-script=openssl.ld -Wl,-Bsymbolic-functions -fPIC shared --prefix=$HOME/ossl-098l
+        make depend
+        make install
```

Comment:
rofl, welp, ok

On Sat, Jan 2, 2016 at 2:47 PM, Paul Kehrer <notifications@github.com>
wrote:

> In .travis/install.sh
> <https://github.com/pyca/cryptography/pull/2553#discussion_r48686914>:
>
> > +    if [[ "${OPENSSL}" == "0.9.8" ]]; then
> > +      # download, compile, and install if it's not already present via travis cache
> > +      # This is 0.9.8l rather than zh because we have some branches for handling
> > +      # < 0.9.8m that won't be exercised with a newer OpenSSL. (RHEL5 is 0.9.8e with
> > +      # patches, but while that's in jenkins we don't get coverage data from it)
> > +      if [[ ! -f "$HOME/ossl-098l/bin/openssl" ]]; then
> > +        curl -O https://www.openssl.org/source/openssl-0.9.8l.tar.gz
> > +        tar zxvf openssl-0.9.8l.tar.gz
> > +        cd openssl-0.9.8l
> > +        echo "OPENSSL_0.9.8L_CUSTOM {
> > +            global:
> > +              *;
> > +        };" > openssl.ld
> > +        ./config no-asm no-ssl2 -Wl,--version-script=openssl.ld -Wl,-Bsymbolic-functions -fPIC shared --prefix=$HOME/ossl-098l
> > +        make depend
> > +        make install
>
> I stand corrected, it looks like 0.9.8l does not properly have parallel
> build support enabled, so -j2 just results in a warning being logged that
> -j1 is used because the jobserver is unavailable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2553/files#r48686914>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084